### PR TITLE
Add suport for /proc/[pid]/task/[tid]/children

### PR DESCRIPTION
**Pull request by zmjackson**
While using this library I came across the need to find the children of a process. I saw that it was not supported, but it seemed fairly easy to implement so I took a shot at it.

I'm new to Rust so I welcome any suggestions.

### CODE REVIEW

File: src/process/task.rs [hunk 1]
```diff
@@ -188,4 +206,30 @@ mod tests {
...
+
+    #[test]
+    fn test_task_children() {
+        // Use tail -f /dev/null to create two infinite processes
+        let mut command = process::Command::new("tail");
+        command.arg("-f").arg("/dev/null");
+        let (mut child1, mut child2) = (command.spawn().unwrap(), command.spawn().unwrap());
+
+        // Cargo tests run on multiple threads under the same process by default
+        // There doesn't seem to be an easy way to get the current thread ID
```
Comment (eminence) [inline src/process/task.rs:218]: What about https://docs.rs/rustix/latest/rustix/thread/fn.gettid.html to get the thread ID?

Comment (zmjackson) [reply src/process/task.rs:218]: Good point. Didn't realize rustix has that utility. I'll change it.

**Comment by eminence** [comment]
You can ignore the CI failure (or you can rebase to get a fix for that)

File: src/process/task.rs [hunk 2]
```diff
@@ -75,11 +76,28 @@ impl Task {
     pub fn schedstat(&self) -> ProcResult<Schedstat> {
         Schedstat::from_reader(FileWrapper::open_at(&self.root, &self.fd, "schedstat")?)
     }
+
+    /// Thread children from `/proc/<pid>/task/<tid>/children`
+    ///
+    /// This data will be unique per task.
```
Comment (eminence) [inline src/process/task.rs:82]: I'd love to see these docs expanded a bit.  In particular, the `procfs(5)` man page lists some caveats that look important and I'd like to see them replicated here.

**Comment by eminence** [comment]
> While using this library I came across the need to find the children of a process. I saw that it was not supported, but it seemed fairly easy to implement so I took a shot at it.

I should also note that using `/proc/[pid]/task/[tid]/children` is not really the recommended way to get the children of a process.   Generally you have to look at each processes parent process and build up a list of children processes from that.  See also [this example](https://github.com/eminence/procfs/blob/master/examples/process_hierarchy.rs)

**Comment by eminence** [comment]
Could you please rebase on top of the latest master branch (instead of merging it into your feature branch)?  This will result is a much cleaner history.  If you're not sure how to do this, I'd be happy to help explain the steps.

**Comment by zmjackson** [comment]
Sorry, I don't have a lot of experience with forked projects. I tried to rebase but clearly it didn't work. What steps should I take to get it in the right state?

**Comment by eminence** [comment]
Oh, I think I see what might have happened.  Correct me if I'm wrong, but I think your successfully rebased, but when you went to push your branch, git told you that it wasn't a fast-forward, and it recommended that you run `git pull`.  Unfortunately, running `git pull` after a rebase is almost always the *wrong* thing to do.  (Not really your fault, I've been annoyed at git for *years* that it continues to make this recommendation)

I hope you don't mind, but I've fixed this for you by force-pushing to your `task-children` branch.  Be sure to fetch and `git reset` your local branch before re-using it.

Since I did a force-push, can you quickly skim the commits in this PR to make sure they look OK to you?

**Comment by zmjackson** [comment]
Sorry for the delay, that is exactly what happened. Thanks so much for the help! Everything looks good.

**Pull request merged**